rasa/nlu/classifiers/keyword_intent_classifier.py

```diff
@@ -1,32 +1,159 @@
-from typing import Any, Optional, Text
+import os
+import logging
+import typing
+import re
+from typing import Any, Dict, Optional, Text
 
+from rasa.nlu import utils
 from rasa.nlu.components import Component
 from rasa.nlu.training_data import Message
 
+logger = logging.getLogger(__name__)
+
+if typing.TYPE_CHECKING:
+    from rasa.nlu.config import RasaNLUModelConfig
+    from rasa.nlu.training_data import TrainingData
+    from rasa.nlu.model import Metadata
+    from rasa.nlu.training_data import Message
+
 
 class KeywordIntentClassifier(Component):
+    """Intent classifier using simple keyword matching.
+
+
+    The classifier takes a list of keywords and associated intents as an input.
+    A input sentence is checked for the keywords and the intent is returned.
+
+    """
 
     provides = ["intent"]
 
-    his = ["hello", "hi", "hey"]
+    defaults = {"case_sensitive": True}
+
+    def __init__(
+        self,
+        component_config: Optional[Dict[Text, Any]] = None,
+        intent_keyword_map: Optional[Dict] = None,
+    ):
+
+        super(KeywordIntentClassifier, self).__init__(component_config)
+
+        self.intent_keyword_map = intent_keyword_map or {}
 
-    byes = ["bye", "goodbye"]
+        self.re_case_flag = (
```

Comment:
```suggestion
        self.case_sensitive = self.component_config["case_sensitive"]
```

Reply:
The `re_case_flag` should actually have been removed. I think it's better to get the config values from the dict, because that's where they are given.

Reply:
in other components we usually create corresponding variables in init, so I think, it makes sense to stick here or change everywhere

Reply:
I thought others had it like this, with the dict. But I just looked at extractors and classifiers and it seems we are just not consistent in this, sometimes even some config is set as attributes while others remain in dict..

Reply:
@tmbo what do you think?

Reply:
@Ghostvv just asked @tmbo and he didn't have a strong opinion on the topic, shall we just go with what it is now?

Reply:
I think either approach is fine and I can't seem to find a good reason to prefer one over the other. But I agree we should be consistent, so let's just go with the one that we already use most @imLew

Reply:
It's really all over the place, I would have to count be sure which we use more 😅